graylog2-web-interface/src/components/lookup-tables/CachesOverview.tsx

```diff
@@ -0,0 +1,145 @@
+        <kbd>guava</kbd> <br />is the same as<br />
+        <kbd>title:guava</kbd>
+      </p>
+    </Popover>
+  );
+};
+
+type Props = {
+  caches: LookupTableCache[],
+  pagination: PaginationType,
+  paginationQueryParameter: PaginationQueryParameterResult,
+};
+
+const CachesOverview = ({ caches, pagination, paginationQueryParameter }: Props) => {
+  const { currentPage, currentPageSize, resetPage } = React.useMemo(() => ({
+    currentPage: paginationQueryParameter.page || 1,
+    currentPageSize: paginationQueryParameter.pageSize || 10,
+    resetPage: paginationQueryParameter.resetPage,
+  }), [paginationQueryParameter]);
+
+  const onPageChange = (newPage: number, newPerPage: number) => {
+    LookupTableCachesActions.searchPaginated(newPage, newPerPage, pagination.query);
+  };
+
+  const onSearch = (query: string, resetLoadingStateCb: () => void) => {
+    resetPage();
+    LookupTableCachesActions.searchPaginated(1, currentPageSize, query).then(resetLoadingStateCb);
+  };
+
+  const onReset = () => {
+    resetPage();
+    LookupTableCachesActions.searchPaginated(currentPage, currentPageSize);
+  };
+
+  return (
+    <Row className="content">
+      <Col md={12}>
+        <h2 style={{ marginBottom: 16 }}>
+          Configured lookup Caches <small>{pagination.total} total</small>
+        </h2>
+        <PaginatedList activePage={currentPage}
+                       pageSize={currentPageSize}
+                       onChange={onPageChange}
+                       totalItems={pagination.total}>
+          <SearchForm onSearch={onSearch} onReset={onReset}>
+            <OverlayTrigger trigger="click" rootClose placement="right" overlay={buildHelpPopover()}>
+              <Button bsStyle="link"
+                      className={Styles.searchHelpButton}>
+                <Icon name="question-circle" fixedWidth />
+              </Button>
+            </OverlayTrigger>
+          </SearchForm>
+          <div style={{ overflowX: 'auto' }}>
+            <Table condensed hover className={Styles.overviewTable}>
+              <thead>
+                <tr>
+                  <th className={Styles.rowTitle}>Title</th>
+                  <th className={Styles.rowDescription}>Description</th>
+                  <th className={Styles.rowName}>Name</th>
+                  <th>Entries</th>
+                  <th>Hit rate</th>
+                  <th>Throughput</th>
+                  <th className={Styles.rowActions}>Actions</th>
+                </tr>
+              </thead>
+              {caches.length === 0
+                ? <Spinner text="Loading caches" />
+                : caches.map((cache: LookupTableCache) => (
+                  <CacheTableEntry key={cache.id} cache={cache} />
+                ))}
+            </Table>
+          </div>
+        </PaginatedList>
+      </Col>
+    </Row>
+  );
+};
+
+export default withPaginationQueryParameter(CachesOverview as React.ComponentType<Props>);
```

Comment:
I think it makes more sense to adjust the type definition for `withPaginationQueryParameter`. If we type it like this:
```
const withPaginationQueryParameter = <C extends React.ComponentType<React.ComponentProps<C>>>(Component: C, obj?: PaginationQueryParameterObject) => {
```

We do not need to use `as` when we use `withPaginationQueryParameter` for a function component.

Reply:
This would also be helpful in `DataAdaptersOverview` and `LookupTablesOverview`.